docs/en/concepts-and-designs/key-principles.md

```diff
@@ -0,0 +1,174 @@
+The Agent Core code would be dynamically added during hybrid compilation, so when the Agent releases new features, 
+users only need to upgrade the Agent enhancement program without modifying the references in the program.
+
+### Code Import
+
+You can see a lot of `imports.go` files anywhere in the SkyWalking Go, such as [imports.go in the root directory](../../../imports.go), but there is no actual code. 
+This is because, during hybrid compilation, if the code to be compiled references other libraries, 
+such as `os`, `fmt`, etc., they need to be referenced through the **importcfg** file during compilation.
+
+The content of the `importcfg` file is shown below, which specifies the package dependency information required for all Go files to be compiled in the current package path.
+
+```
+packagefile errors=/var/folders/wz/s5m922z15vz4fjhf5l4458xm0000gn/T/go-build2774248373/b006/_pkg_.a
+packagefile internal/itoa=/var/folders/wz/s5m922z15vz4fjhf5l4458xm0000gn/T/go-build2774248373/b027/_pkg_.a
+packagefile internal/oserror=/var/folders/wz/s5m922z15vz4fjhf5l4458xm0000gn/T/go-build2774248373/b035/_pkg_.a
+```
+
+So when the file is copied and added to the compilation process, the relevant dependency libraries need to be declared in `importcfg`. 
+Therefore, by predefining `import` in the project, the compiler can be forced to introduce the relevant libraries during compilation, 
+thus completing the dynamic enhancement operation.
+
+## Plugin with Agent Core
+
+As mentioned in the previous section, it is not possible to dynamically add dependencies between modules. 
+Agent can only modify the `importcfg` file to reference dependencies if we are sure that the previous dependencies have already been loaded, 
+but this is often impractical. For example, Agent cannot introduce dependencies from the plugin code into the Agent Core, 
+because the plugin is unaware of the Agent's existence. This raises a question: how can agent enable communication between plugins and Agent Core?
+
+Currently, agent employ the following method: a global object is introduced in the `runtime` package, provided by Agent Core. 
+When a plugin needs to interact with Agent Core, it simply searches for this global object from `runtime` package. The specific steps are as follows:
+
+1. **Global object definition**: Add a global variable when the `runtime` package is loaded and provide corresponding set and get methods.
+2. **Set the variable when the Agent loads**: When the Agent Core is copied and enhanced, import the method for setting the global variable and initialize the object in the global variable.
+3. **Plugin enhancement**: When the plugin is enhanced, import the method for getting the global variable and the interface definition for the global variable. 
+At this point, we can access the object set in Agent Core and use the defined interface for the plugin to access methods in Agent Core.
```

Comment:
```suggestion
At this point, we can access the object set in Agent Core and use the defined interface for the plugin to access methods in Agent Core.
```

I think we should only access Agent Core APIs? what does this mean?

Reply:
Yes, the plugin can only access the Agent Core API. Then, the Agent Core API would utilize the global variables in the runtime package to operate the Agent Core.